src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -253,6 +254,76 @@ public IActionResult Recover([FromBody]WalletRecoveryRequest request)
             }
         }
 
+        /// <summary>
+        /// Recovers a wallet using only the extended public key.
+        /// </summary>
+        /// <param name="request">The object containing the parameters used to recover a wallet.</param>
+        /// <returns></returns>
+        [Route("recover-via-extpubkey")]
+        [HttpPost]
+        public IActionResult RecoverViaExtPubKey([FromBody]WalletExtPubRecoveryRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+
+            if (!this.ModelState.IsValid)
+            {
+                return BuildErrorResponse(this.ModelState);
+            }
+
+            try
+            {
+                string accountExtPubKey = this.SanitiseExtPubKey(request.ExtPubKey);
+
+                Wallet wallet = this.walletManager.RecoverWallet(request.Name, ExtPubKey.Parse(accountExtPubKey), request.AccountIndex, request.CreationDate);
+
+                // start syncing the wallet from the creation date
+                this.walletSyncManager.SyncFromDate(request.CreationDate);
+
+                return this.Ok();
+            }
+            catch (WalletException e)
+            {
+                // indicates that this wallet already exists
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.Conflict, e.Message, e.ToString());
+            }
+            catch (FileNotFoundException e)
+            {
+                // indicates that this wallet does not exist
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.NotFound, "Wallet not found.", e.ToString());
+            }
+            catch (Exception e)
+            {
+                this.logger.LogError("Exception occurred: {0}", e.ToString());
+                return ErrorHelpers.BuildErrorResponse(HttpStatusCode.BadRequest, e.Message, e.ToString());
+            }
+        }
+
+        private string SanitiseExtPubKey(string requestExtPubKey)
+        {
+            if (this.network.IsBitcoin())
+                return requestExtPubKey;
+
+            byte[] stratisVersionBytes = network.GetVersionBytes(Base58Type.EXT_PUBLIC_KEY, true); 
+            byte[] ledgerVersionByte = new byte[] { (0x04), (0x88), (0xC2), (0x1E) };
+
+            byte[] decodedExtPubKey = Encoders.Base58Check.DecodeData(requestExtPubKey);
+            byte[] extPubKeyVersionBytes = decodedExtPubKey.Take(4).ToArray();
+
+            if (extPubKeyVersionBytes.SequenceEqual(stratisVersionBytes))
+            {
+                return requestExtPubKey;
+            }
+            else if (extPubKeyVersionBytes.SequenceEqual(ledgerVersionByte))
+            {
+                byte[] vchData = decodedExtPubKey.Skip(4).ToArray();
```

Comment:
What is vchData and why 4 bytes are skipped? Please add comments.

Reply:
@bokobza what does vch mean?

Reply:
I think it's hungarian for vector of characters (character array).

Reply:
@mikedennis is right. vch is hungarian notation for vector of char. But here instead of char or uchar you use bytes so the name is misleading.

Reply:
ok cool, I will correct it